src/jquery.multiselect.filter.js

```diff
@@ -181,7 +181,7 @@
       // show/hide optgroups
       $groups.each(function() {
         var $this = $(this);
-        if (!$this.children('li').filter(':visible').length)
+        if (!$this.children(':not(:first-child)').filter(':visible').length)
```

Comment:
For proper nesting, I think you need to find the li's that are descendants of the current li containing the anchor.

Reply:
The [accepted](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li) children of an `li` is any [flow content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Flow_content)